Use mkinitramfs only if ramdisk is not set in /etc/kernel-img.conf.
--- a/dkms
+++ b/dkms
@@ -215,15 +215,13 @@ function remake_initrd()
                        -e "s/['\"]*//g" \
                        /etc/kernel-img.conf | \
                    awk '{print $1}')
-    else
-        if type update-initramfs >/dev/null 2>&1; then
-            mkinitrd='update-initramfs'
-        fi
     fi
 
     # Support initramfs distributions (Ubuntu).
-    if [ -x "/usr/sbin/mkinitramfs" ]; then
-        mkinitrd='mkinitramfs'
+    if [-z "$mkinitrd" ] || [ "$mkinitrd" == "mkinitrd" ]; then
+	    if [ -x "/usr/sbin/mkinitramfs" ]; then
+		    mkinitrd='mkinitramfs'
+	    fi
     fi
 
     $mkinitrd --version >/dev/null 2>&1
